src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmServiceImpl.java

```diff
@@ -313,15 +327,21 @@ private void populateGermplasmPedigree(final List<GermplasmDTO> germplasmDTOList
 	}
 
 	@Override
-	public List<AttributeDTO> getAttributesByGid(
-		final String gid, final List<String> attributeDbIds, final Integer pageSize, final Integer pageNumber) {
-		this.validateGidAndAttributes(gid, attributeDbIds);
-		return this.germplasmDataManager.getAttributesByGid(gid, attributeDbIds, pageSize, pageNumber);
+	public List<AttributeDTO> getAttributesByGermplasmGUID(
+			final String germplasmGUID, final List<String> attributeDbIds, final Integer pageSize, final Integer pageNumber) {
+		this.validateGuidAndAttributes(germplasmGUID, attributeDbIds);
+		final Optional<GermplasmDTO> germplasmDTO = this.germplasmService.getGermplasmDTOByGUID(germplasmGUID);
+		if (germplasmDTO.isPresent()) {
+			return this.germplasmDataManager.getAttributesByGid(germplasmDTO.get().getGid(), attributeDbIds, pageSize, pageNumber);
+		} else {
+			throw new ApiRuntimeException("Invalid Germplasm Id");
+		}
+
 	}
 
 	@Override
-	public long countAttributesByGid(final String gid, final List<String> attributeDbIds) {
-		return this.germplasmDataManager.countAttributesByGid(gid, attributeDbIds);
+	public long countAttributesByGermplasmGUID(final String germplasmGUID, final List<String> attributeDbIds) {
+		return this.germplasmDataManager.countAttributesByGid(germplasmGUID, attributeDbIds);
```

Comment:
Should we not call first: `this.germplasmService.getGermplasmDTOByGUID` to get the GID to pass to `countAttributesByGid`?